partner_credit_limit/migrations/11.0.0.0.2/post-migration.py

```diff
@@ -0,0 +1,19 @@
+from odoo import api, SUPERUSER_ID
+
+
+def migrate(cr, version):
+    set_credit_limit_values_to_partner(cr)
+
+
+def set_credit_limit_values_to_partner(cr):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    env.cr.execute(
+        """
+        SELECT id, credit_limit FROM res_partner
+        WHERE credit_limit IS NOT NULL""")
+    data = env.cr.dictfetchall()
+    credit_limit_vals = {p['id']: p['credit_limit'] for p in data}
+    partner_ids = [p.get('id') for p in data]
+    partners = env['res.partner'].browse(partner_ids or [])
+    for partner in partners:
+        partner.write({'credit_limit': credit_limit_vals.get(partner.id, 0)})
```

Comment:
You should force the write for all companies
I mean, a
for company in companies:
  partner.with_context(force_company=company).write(...

I don't remember the real flag

Since that old version the same value applies for all companies

Reply:
You are moving to company_dependent 2 fields
But you are updating just one

What about “grace_payment_days”?

Reply:
grace_payments does not appear on DB yet, @edy1192 still debugging

Reply:
Hi @moylop260 @hugho-ad 

I think what happens is that when the module is installed with the change of company_dependent, these fields become computed and are not found in the bd. But if the module was installed before this change, it already exists in bd. It is the theory that I have, but I have to do other tests to verify this.

But right now I am with other issues.

Thanks for your comments.

Regards.

Reply:
> You should force the write for all companies
> I mean, a
> for company in companies:
> partner.with_context(force_company=company).write(...
> 
> I don't remember the real flag
> 
> Since that old version the same value applies for all companies

Got it, I will try that. Thanks 😄

Reply:
@edy1192 

grace_payments only will found on instances with partner_credit_limit module previously installed only because these fields become computed ones.

So don't worry, the migration script only will be executed on those instances

try it and let us know

Reply:
Thanks @hugho-ad  you're right! 😄 

@moylop260 You were right too! Without the force company it only established the values in one company and not in all of them. Thanks for the advice 😄 

Regards.